src/software/ai/evaluation/calc_best_shot.h

```diff
@@ -60,207 +38,18 @@ std::optional<Shot> calcBestShotOnGoal(const Point &goal_post_neg,
  * @param field The field
  * @param friendly_team The friendly team
  * @param enemy_team The enemy team
- * @param point The point that the shot will be taken from
+ * @param shot_origin The point that the shot will be taken from
+ * @param goal The goal to shoot at
  * @param radius The radius for the robot obstacles
  * @param robots_to_ignore The robots to ignore
- * @param shoot_at_enemy_goal Whether or not the shot is at the enemy goal. If true,
- * the shot will be treated as a shot at the enemy goal, and if false, the shot will
- * be treated as a shot at the friendly goal
  *
  * @return the best target to shoot at and the largest open angle interval for the
  * shot (this is the total angle between the obstacles on either side of the shot
  * vector). If no shot can be found, returns std::nullopt
  */
 
 std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_team,
-                                       const Team &enemy_team, const Point &point,
-                                       bool shoot_at_enemy_goal,
+                                       const Team &enemy_team, const Point &shot_origin,
+                                       TeamType goal,
                                        double radius = ROBOT_MAX_RADIUS_METERS,
                                        const std::vector<Robot> &robots_to_ignore = {});
```

Comment:
can we switch places of these arguments, so users don't have to pass in `ROBOT_MAX_RADIUS` to give `robots_to_ignore`?

Reply:
Done.